keras_cv/ops/box_matcher.py

```diff
@@ -0,0 +1,235 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"Argmax-based box matching"
+
+from typing import List
+from typing import Tuple
+
+import tensorflow as tf
+
+
+class ArgmaxBoxMatcher:
+    """Box matching logic based on argmax of highest value (e.g., iou).
+
+    This class computes matches from a similarity matrix. Each row will be
+    matched to at least one column, the matched result can either be positive
+     / negative, or simply ignored depending on the setting.
+
+    The settings include `thresholds` and `match_values`, for example if:
+    1) thresholds=[negative_threshold, positive_threshold], and
+       match_values=[negative_value, ignore_value, positive_value]: the rows will
+       be assigned to positive_value if its argmax result is above
+       positive_threshold; the rows will be assigned to negative_value if its
+       argmax result is below negative_threshold, and the rows will be assigned
+       to ignore_value if its argmax result is between negative_threshold and
+       positive_threshold.
+    2) thresholds=[negative_threshold, positive_threshold], and
```

Comment:
I have little difficulty to parse this example, comparing to example 1. The only difference I can see is that the order of the  ignore_value and negative_value are switched, but its unclear to me why they are different. 

To me, a more clear description is that:
thresholds is buckets, so for [t1, t2], there will be three buckets: (-inf, t1], (t1, t2], (t2, +inf), and user will provide a list of 3 values for each of the bucket. You can have arbitrary number of buckets (skip the logic of positive and negative).

Reply:
positive / ignore is a OD specific concept, and our example should reflect that.
Though this does present some confusion, so I updated the examples to reflect how 1) and 2) are different.